ssl/ssl_lib.c

```diff
@@ -2599,7 +2599,8 @@ ossl_ssize_t SSL_sendfile(SSL *s, int fd, off_t offset, size_t size, int flags)
             BIO_set_retry_write(sc->wbio);
         else
 #endif
-            ERR_raise(ERR_LIB_SSL, SSL_R_UNINITIALIZED);
+            ERR_raise_data(ERR_LIB_SYS, get_last_socket_error(),
+                           "ktls_sendfile failure");
```

Comment:
Hmm. The code above uses `get_last_sys_error`, but this uses `get_last_socket_error` which looks strange.

Reply:
If I understand difference between `GetLastError()` (`get_last_sys_error()`) and `WSAGetLastError()` (`get_last_socket_error()`) right, then we really should be using `get_last_socket_error()` here so we can obtain error code from socket subsystem on windows. The lines above should be changed as well, I think. Perhaps it should be done in seperate PR.

The `get_last_sys_error()` vs. `get_last_socket_error()` only matters on windows. On other OSes both macros are alias of `errno`.

Reply:
@Sashan is right, on linux it should be the same, since their both aliased to errno, but you're right @mattcaswell it should be get_last_sys_error for consistency

Reply:
updated to use get_last_sys_error, and added a simmilar error for SSL_write

Reply:
I am not sure this kind of changes should be done in stable releases. Especially the change for SSL_write().

Reply:
you think it should be targeted for the next minor release?  Or is this a major-only change?

Reply:
IMO OK for a minor release.

Reply:
Ok, will propose for 3.4